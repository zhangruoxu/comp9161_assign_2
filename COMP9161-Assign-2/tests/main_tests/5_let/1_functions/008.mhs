main = 
  let g = letfun f x = 
            if True 
              then (let h = letfun g y = x; in h) 
              else (let h = letfun g y = (x+1); in h) ; 
  in g;
