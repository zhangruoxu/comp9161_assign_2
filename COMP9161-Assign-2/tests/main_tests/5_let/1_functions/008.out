main :: forall k. (Int -> k -> Int) = let g :: forall j. (Int -> j -> Int) = (letfun f :: (Int -> j -> Int) x = if True
                                                                                                                   then let h :: forall c. (c -> Int) = (letfun g :: (c -> Int) y = x);
                                                                                                                        in h
                                                                                                                   else let h :: forall f. (f -> Int) = (letfun g :: (f -> Int) y = (+) x 1);
                                                                                                                        in h);
                                      in g;
